codec/src/main/java/io/netty/handler/codec/MessageToByteEncoder.java

```diff
@@ -46,7 +46,7 @@
 public abstract class MessageToByteEncoder<I> extends ChannelOutboundHandlerAdapter {
 
     private final TypeParameterMatcher matcher;
-    private final boolean preferDirect;
+    protected final boolean preferDirect;
```

Comment:
@jasobrown - if this method existing in the base class you would just call it and wouldn't need to directly access `preferDirect` (or know how it is determined).

```java
Lz4FrameEncoder.allocateBuffer(ctx, msg)
// which calls
Lz4FrameEncoder.allocateBuffer(ctx, msg,
        /*MessageToByteEncoder has the responsibility to calculate the preferDirect*/)
```